test/tests.ts

```diff
@@ -507,4 +507,55 @@ describe('Specifying dockerfiles', () => {
 			countedResolve();
 		});
 	});
+
+	it('should allow a Dockerfile to be specified in a different location', () => {
```

Comment:
You can have async test functions.

Constructing promises with an `async` function feels wrong: `return new Promise(async (resolve, reject) => {`